### PR TITLE
Copy EntityListFilterBar and Supporting Components into Prototype; closes #2017

### LINKED ISSUE #2017: Copy EntityListFilterBar and Supporting Components into Prototype
In EDTR v1, the filter bars for the DatesList and TicketsList were composed of entity specific filters added to a set of base filters located in 

```assets/ZZZ/components/ui/entity-list-filter-bar```

The contents of this folder need to be copied over to:

```
assets/prototype/application/ui/components/display
```

and then refactored to:
- utilize new data entities 
- use dependencies from EDTR v2 (prototype)
- update all other imports as necessary

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
Copy as described here: https://github.com/eventespresso/event-espresso-core/issues/2017

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/datetimes/dateList/DateList.js [hunk 1]
```diff
@@ -47,17 +47,17 @@ const DateList = ({ datetimes, tickets }) => {
...
+		setListView,
+		setPerPage: () => null,
+		setSearchText: () => null,
 	};
 
 	return (
 		<div>
 			{header}
-			<EntityListFilterBar {...entityListFilterBarProps} />
+			<DatesListFilterBar {...datesListFilterBarProps} />
```
Comment (manzoorwanijk) [inline assets/prototype/domain/eventEditor/datetimes/dateList/DateList.js]: Nice 👍

File: assets/prototype/domain/eventEditor/datetimes/dateList/filter/DatesListEntityFilters.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * @param {string} datesSortedBy
+ * @param {string} displayDates
+ * @return {Object} EditorDatesListView with added DateListFilters
+ */
+const DatesListEntityFilters = ({ showDates, datesSortedBy, displayDates }) => {
+	const { setShowDates, setDatesSortedBy, setDisplayDates } = {
+		setShowDates: () => null,
+		setDatesSortedBy: () => null,
+		setDisplayDates: () => null,
+	};
```
Comment (manzoorwanijk) [inline assets/prototype/domain/eventEditor/datetimes/dateList/filter/DatesListEntityFilters.js]: These will come from the filter state manager, right?

Comment (nicolad) [reply assets/prototype/domain/eventEditor/datetimes/dateList/filter/DatesListEntityFilters.js]: Absolutely, this is why I passed temp `() => null`

**Comment by nicolad** [comment]
This PR has been merged into https://github.com/eventespresso/event-espresso-core/pull/2014

**Pull request merged**